src/back/spv/writer.rs

```diff
@@ -1172,6 +1172,11 @@ impl Writer {
             }
             crate::Binding::BuiltIn(built_in) => {
                 use crate::BuiltIn as Bi;
+                let is_flat = match built_in {
+                    Bi::Position { .. } => false,
+                    _ => true,
+                };
```

Comment:
The actual rule from the Vulkan spec is:

> - VUID-StandaloneSpirv-Flat-04744
>    Any variable with integer or double-precision floating-point type and with Input storage class in a fragment shader, must be decorated Flat

It's true that `spirv-val` doesn't complain about (say):

```
@fragment
fn fs_main(@location(0) index: u32) -> @location(0) vec4<f32> {
    return vec4<f32>(0.0, 0.0, 0.0, 0.0);
}
```

but since the background story is that these validations are just being implemented piecemeal as somebody's interest leads them, I think we should assume the above might well stop passing validation at some point in the future.

Anyway, that is too many words to say: Let's implement the rule as expressed in the Vulkan spec and attach `Decoration::Flat` to everything whose type is `Sint`, `Uint`, or 64-bit `Float`.